flashinfer/moe_ep/kernel_src/sm120/split_cutedsl_megakernel/src/moe_sm120_mxfp4mxfp8_split/kernel_fc2_combine_n16_dual.py

```
@@ -0,0 +1,1144 @@
+        # never wait on rank progress or on another CTA's admission.
+        if cutlass.const_expr(green_trace is not None):
+            if warp_idx == cutlass.Int32(0):
+                if lane_idx == cutlass.Int32(0):
+                    cute.arch.store(green_trace.iterator + trace_word(trace_role, cta_linear_id, FIELD_KERNEL_EXIT), read_globaltimer(), scope='gpu')
+
+    # The shared host launcher calls this base-class hook.  K2 binds it to the
+    # physically separate FC2-only device body above.
+    fc1fc2_kernel_impl = fc2_combine_kernel_impl
+
+
+__all__ = ["Sm120Fc2CombineKernel"]
```

Ruff (0.16.2)

[error] Undefined name `Sm120Fc2CombineKernel` in `__all__`

(F822)